addons/project/models/project.py

```diff
@@ -2125,6 +2138,37 @@ def _unlink_except_recurring(self):
             # TODO: show a dialog to stop the recurrence
             raise UserError(_('You cannot delete recurring tasks. Please disable the recurrence first.'))
 
+    @api.model
+    def _read_group_user_ids(self, users, domain, order):
+        """
+        _read_group_user_ids allow to display empty rows in the Gantt view of a given project (user with open task assigned in the current project)
+        """
+        # 1. filter domain to remove planned_date_begin and planned_date_ending
+        filtered_domain = [dom for dom in domain if isinstance(dom, list) or isinstance(dom, tuple) and len(dom) == 3 and not dom[0] in ['planned_date_begin', 'planned_date_end']]
+
+        # 2. Check if the Gantt view is only related to a single project (if not, the function do nothing and return 'users')
+        display_project_id = False
+        for dom in filtered_domain:
+            if dom[0] == 'display_project_id':
+                if dom[1] in ['=', 'ilike']:
+                    display_project_id = dom[2]
+                elif dom[1] == 'in':
+                    converted_dom = dom[2] if isinstance(dom[2], set) or isinstance(dom[2], list) or isinstance(dom[2], tuple) else json.loads(dom[2]) #converting string representation of a list to a list
+                    if isinstance(converted_dom, list) or isinstance(converted_dom, tuple) or isinstance(converted_dom, set) and len(converted_dom) >= 2:
+                        return users
+                    elif isinstance(converted_dom, list) or isinstance(converted_dom, tuple) or isinstance(converted_dom, set) and len(converted_dom) == 1:
+                        if display_project_id:
+                            return users
+                        else:
+                            display_project_id = converted_dom[0]
+        if not display_project_id:
+            return users
```

Comment:
Not again :sweat_smile: 
But I think we can just use the `default_project_id` key of the context. Not sure really.

Reply:
The logic I used was not entirely adequate. I've rewritten it in a (I think) better way. To be short:
1- I first check the key default_project_id (set if I'm in the view of a project).
2- Then I check the domain is the key is not set to see tasks of which project are displayed.